src/components/ImportAction.tsx

```diff
@@ -0,0 +1,78 @@
+import * as React from 'react';
+import { useDispatch } from 'react-redux';
+import { useTranslation } from 'react-i18next';
+import { useHistory, useLocation } from 'react-router-dom';
+import { Button, ButtonVariant, Tooltip } from '@patternfly/react-core';
+import { PlusCircleIcon } from '@patternfly/react-icons';
+import { useActiveNamespace } from '@openshift-console/dynamic-plugin-sdk';
+import { closeOLS, importCodeblock } from '../redux-actions';
+import ConfirmationModal from './ConfirmationModal';
+
+type Props = {
+  value: string;
+};
+
+const ImportAction: React.FC<Props> = ({ value }) => {
+  const { t } = useTranslation('plugin__lightspeed-console-plugin');
+  const location = useLocation();
+  const dispatch = useDispatch();
+  const history = useHistory();
+  const [activeNamespace] = useActiveNamespace();
+
+  const isImportYAMLPage = location?.pathname?.split('/').pop() === 'import';
+
+  const triggerRef = React.createRef<HTMLButtonElement>();
+  const [showModal, setShowModal] = React.useState(false);
+
+  const generateRandomNumericString = (length = 4): string => {
+    const digits = '0123456789';
+    return Array.from({ length }, () => digits[Math.floor(Math.random() * digits.length)]).join('');
+  };
+
+  const handleRedirect = (event: React.MouseEvent<HTMLButtonElement>) => {
+    if (event.currentTarget.id === 'leave') {
+      history.push(`/k8s/ns/${activeNamespace}/import?ols=true`);
+    }
+    setShowModal(false);
+    dispatch(closeOLS());
+  };
+
+  const handleClick = () => {
+    if (isImportYAMLPage) {
+      history.replace(`${location.pathname}?ols=true`);
+    } else {
+      setShowModal(true);
+    }
+    dispatch(
+      importCodeblock({
+        value,
+        id: generateRandomNumericString(),
```

Comment:
Actually, the codeblock ID is being used indirectly in the useEffect hook of the console. This allows us to paste the contents of the codeblock to the editor even if the user selects the same codeblock. This is required to ensure that all edge cases are properly handled.

Reply:
I'm still not sure where the ID is actually used on the console side, but could you please add a comment in the code here to explain why it's needed? It seems like something that could confuse someone looking at the code in the future.

Reply:
Also, if this just needs to be a unique ID, you could use Lodash like `_.uniqueId('ImportCodeBlock_')`. This would be a bit clearer and would also ensure uniqueness.

Reply:
Thanks for the suggestion. It's working as expected. Have used the lodash method and also have added a comment for future development.